ts/twitter/src/app/posts/[id]/_components/post-detail.tsx

```diff
@@ -1,27 +1,41 @@
 "use client";
 
+import { useColorMode } from "@/lib/components/ui/color-mode";
 import type { Post } from "@/lib/models/post";
 import { Avatar, Box, Button, Flex, Text } from "@chakra-ui/react";
 import { useState } from "react";
 import { mockPost } from "../mocks/post";
 
 export const PostDetail = () => {
+  const { colorMode } = useColorMode();
   const post: Post = mockPost;
   const [isFollowing, setIsFollowing] = useState<boolean>(false);
   const toggleFollowState = () => setIsFollowing((prev) => !prev);
 
   return (
     <Box key={post.id} p={8} borderWidth={1} borderRadius="md">
       <Flex>
-        <Avatar size="md" name={post.user_id} mr={2} />
-        <Text fontWeight="bold" fontSize="xl" mr={4} mt={2}>
-          {post.user_id}
-        </Text>
-        <Button borderRadius="full" onClick={toggleFollowState}>
-          {/* TODO: https://github.com/okuda-seminar/Twitter-Clone/issues/452 
+        <Avatar.Root size="xl">
+          <Avatar.Fallback name={post.user_id} />
+          <Avatar.Image />
+        </Avatar.Root>
+        <Flex flex="1" alignItems="center" ml={2}>
+          <Text fontWeight="bold" fontSize="xl">
+            {post.user_id}
+          </Text>
+          <Button
+            borderRadius="full"
+            ml={2}
+            onClick={toggleFollowState}
+            fontSize="md"
+            bg="gray.100"
+            color={colorMode === "dark" ? "white" : "black"}
```

Comment:
```suggestion
            color={useColorModeValue("black", "white")}
```
You can use `useColorModeValue`.